mesonbuild/modules/python.py

```diff
@@ -527,19 +529,21 @@ def __init__(self, python: 'PythonExternalProgram', interpreter: 'Interpreter'):
             'path': self.path_method,
         })
 
-    @permittedKwargs(mod_kwargs)
-    def extension_module_method(self, args: T.List['TYPE_var'], kwargs: 'TYPE_kwargs') -> 'SharedModule':
-        if 'install_dir' in kwargs:
-            if 'subdir' in kwargs:
-                raise InvalidArguments('"subdir" and "install_dir" are mutually exclusive')
-        else:
-            subdir = kwargs.pop('subdir', '')
-            if not isinstance(subdir, str):
-                raise InvalidArguments('"subdir" argument must be a string.')
-
-            kwargs['install_dir'] = self._get_install_dir_impl(False, subdir)
-
-        new_deps = mesonlib.extract_as_list(kwargs, 'dependencies')
+    @typed_pos_args('python.extension_module', str, varargs=(str, mesonlib.File, build.CustomTarget, build.CustomTargetIndex, build.GeneratedList))
+    @typed_kwargs('python.extension_module', *_MOD_KWARGS)
+    def extension_module_method(self, args: T.Tuple[str, T.List[BuildTargetSource]],
+                                kwargs: ExtensionModuleKw) -> 'SharedModule':
+        # Create a copy of the keyword arguments, but as a SharedModule acrugment dictionary
+        n_kwargs = T.cast('BuildTarget', kwargs.copy())
+        del n_kwargs['subdir']  # type: ignore
+
+        if kwargs['install_dir'] and kwargs['subdir']:
```

Comment:
Isn't this missing `is not None`? Can install_dir be False and/or subdir be '' ?

Reply:
Actually, no. 'subdir' defaults to '', and 'install_dir' defaults to `[]`, so checking for `None` wont work.